### PR TITLE
Add Action0 and Action1 that will be able to be made null-safe

### PR DESCRIPTION
**Pull request by evanweible-wf**

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: lib/src/action.dart [hunk 1]
```diff
@@ -20,11 +20,30 @@ import 'package:w_common/disposable.dart';
...
+  @override
+  String get disposableTypeName => 'Action1';
+
+  @override
+  Future call([void payload]) => super.call(null);
+}
+
+/// Like [Action1], but payloads cannot be made non-nullable since the argument
+/// to [call] is optional.
+class Action<T> extends Action1<T> {
```
Comment (evanweible-wf) [inline lib/src/action.dart:34]: My thought is that we'd either:
- keep this type around as a common base type, but could change it to an interface
- use Action1 as the common base type where needed

File: lib/src/action.dart [hunk 2]
```diff
@@ -20,11 +20,30 @@ import 'package:w_common/disposable.dart';
 
 import 'package:w_flux/src/constants.dart' show v3Deprecation;
 
+/// Like [Action1], but without an expected payload (aka a void payload).
+class Action0 extends Action1<void> {
+  @override
+  String get disposableTypeName => 'Action1';
```
Comment (sorenthompson-wk) [inline lib/src/action.dart:26]: should this be 'Action0'

File: lib/src/action.dart [hunk 3]
```diff
@@ -20,11 +20,30 @@ import 'package:w_common/disposable.dart';
 
 import 'package:w_flux/src/constants.dart' show v3Deprecation;
 
+/// Like [Action1], but without an expected payload (aka a void payload).
+class Action0 extends Action1<void> {
```
Comment (greglittlefield-wf) [inline lib/src/action.dart:24]: We may want to consider making this `extends Action<Null>` instead, to help prevent consumers from accidentally passing in values without knowing they'll be discarded.

Example: https://dartpad.dev/?id=97af2f0dd8c7919807bd6864ab085794

The relevant pieces from that ^
```dart
main() {
  {
    final action = VoidAction0();
    action();     
    action(null);
    action('foo');
    //     ^^^^^
    // No error or other indication that there's a potential issue with 
    // this usage.
    //
    // The value being passed in is discarded, but the consumer may not know that.
  }

  {
    final action = NullAction0();
    action();
    action(null);
    action('foo'); 
    //      ^^^^^
    // Error: The argument type 'String' can't be assigned to the parameter type 'Null'.
  }
}

class VoidAction0 extends Action1<void> {
  @override
  Future call([void payload]) => super.call(null);
}

class NullAction0 extends Action1<Null> {
  @override
  Future call([Null payload]) => super.call(null);
}
```

File: lib/src/store.dart [hunk 4]
```diff
@@ -153,16 +153,31 @@ class Store extends Stream<Store> with Disposable {
   /// called until that future has resolved.
   ///
   /// If the `Store` has been disposed, this method throws a [StateError].
-  void triggerOnActionV2<T>(Action<T> action,
-      [FutureOr<dynamic> onAction(T payload)]) {
-    if (isOrWillBeDisposed) {
-      throw StateError('Store of type $runtimeType has been disposed');
-    }
+  @deprecated
```
Comment (greglittlefield-wf) [inline lib/src/store.dart:156]: Was there a reason this was deprecated and triggerOnAction1 added, other than cleanliness and improved naming consistency? Those seem like good enough reasons, but I was curious if there was some other factor

**Review by greglittlefield-wf — COMMENTED** [review]
Just one comment and question, otherwise the approach generally looks good to me, and I can't think of any issues or potential issues that aren't also present in the other approach (`Action`/`Action2`).

File: lib/src/store.dart (hunk 4 above)
Comment (evanweible-wf) [reply lib/src/store.dart:156]: Nope that was it

**Comment by evanweible-wf** [comment]
Talked through this as a team and decided that this still isn't good enough to justify the change/extra work, so we'll stick with https://github.com/Workiva/w_flux/pull/188

**Pull request closed without merging**